File: rahila_labs_rider_app/lib/models/rider.dart
class Rider {
  final int id;
  final String name;
  final String email;
  final String phone;
  final String availabilityStatus;
  final double? gpsLatitude;
  final double? gpsLongitude;
  final String? profilePhoto;
  final RiderStats? stats;

  Rider({
    required this.id,
    required this.name,
    required this.email,
    required this.phone,
    required this.availabilityStatus,
    this.gpsLatitude,
    this.gpsLongitude,
    this.profilePhoto,
    this.stats,
  });

  factory Rider.fromJson(Map<String, dynamic> json) {
    return Rider(
      id: json['id'],
      name: json['name'],
      email: json['email'],
      phone: json['phone'],
      availabilityStatus: json['availability_status'] ?? 'offline',
      gpsLatitude: json['gps_latitude']?.toDouble(),
      gpsLongitude: json['gps_longitude']?.toDouble(),
      profilePhoto: json['profile_photo'],
      stats: json['stats'] != null ? RiderStats.fromJson(json['stats']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'email': email,
      'phone': phone,
      'availability_status': availabilityStatus,
      'gps_latitude': gpsLatitude,
      'gps_longitude': gpsLongitude,
      'profile_photo': profilePhoto,
    };
  }
}

class RiderStats {
  final int completedTasks;
  final int pendingTasks;

  RiderStats({
    required this.completedTasks,
    required this.pendingTasks,
  });

  factory RiderStats.fromJson(Map<String, dynamic> json) {
    return RiderStats(
      completedTasks: json['completed_tasks'] ?? 0,
      pendingTasks: json['pending_tasks'] ?? 0,
    );
  }
}

File: backend/app/models/rider.py
from datetime import datetime, timezone
from .base import db


class Rider(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(100), nullable=False)
    email = db.Column(db.String(120), unique=True, nullable=False)
    phone = db.Column(db.String(20), nullable=False)
    password_hash = db.Column(db.String(255), nullable=False)

    # Location tracking
    gps_latitude = db.Column(db.Float)
    gps_longitude = db.Column(db.Float)
    last_location_update = db.Column(db.DateTime)

    # Status and profile
    availability_status = db.Column(db.String(20), default='available')  # available, busy, offline
    profile_photo = db.Column(db.String(255))

    # Timestamps
    created_at = db.Column(db.DateTime, default=lambda: datetime.now(timezone.utc))
    updated_at = db.Column(db.DateTime, default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc))

    def to_dict(self):
        return {
            'id': self.id,
            'name': self.name,
            'email': self.email,
            'phone': self.phone,
            'gps_latitude': self.gps_latitude,
            'gps_longitude': self.gps_longitude,
            'last_location_update': (
                self.last_location_update.isoformat() if self.last_location_update else None
            ),
            'availability_status': self.availability_status,
            'profile_photo': self.profile_photo,
            'created_at': self.created_at.isoformat(),
        }
